Source/FullScreenPokemon.ts

```diff
@@ -1355,6 +1357,25 @@ module FullScreenPokemon {
             thing.cycling = true;
             thing.speedOld = thing.speed;
             thing.speed = this.MathDecider.compute("cycleSpeed", thing);
+
+            repeats = thing.FSP.getCharacterWalkingInterval(thing);
+            distance = repeats * thing.speed;
+
+            thing.walking = true;
+            thing.FSP.animateCharacterSetDirection(thing, thing.direction);
+            thing.FSP.animateCharacterSetDistanceVelocity(thing, distance);
```

Comment:
I don't think this logic should be here. In my understanding, cycling immediately changes the visual attribute of the player when it starts, and increases the speed they _will_ be when they start walking.